src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -635,32 +635,28 @@ private void acknowledgementRoutes(){
     }
     private void paybillRoutes()
     {
-        from("rest:POST:/accounts/validate/{primaryIdentifierName}/{primaryIdentifierVal}")
+        from("rest:POST:/accounts/validate/{secondaryIdentifierName}/{secondaryIdentifierVal}")
                 .id("validation-ams")
-                .log(LoggingLevel.INFO, "Validation Check for identifier type paygops")
+                .log(LoggingLevel.INFO, "Validation Check")
                 .process(e -> {
                     String paybillRequestBodyString = e.getIn().getBody(String.class);
                     logger.info("Payload : {}",paybillRequestBodyString);
                     JSONObject body = new JSONObject(paybillRequestBodyString);
-                    String amsURL="";
-                    // Finding the AMS connector
-                    String finalAmsVal = amsUtils.getAMSName(body);
-                    if(finalAmsVal.equalsIgnoreCase("paygops"))
-                    {
-                        amsURL=paygopsHost;
-                    }
-                    else if(finalAmsVal.equalsIgnoreCase("roster"))
-                    {
-                        amsURL=rosterHost;
-                    }
-                    logger.debug("Final Value for ams : " + finalAmsVal);
-                    logger.debug("AMS URL : {}",amsURL);
+                    String amsURL = e.getIn().getHeader("amsUrl").toString();
```

Comment:
This URL header should be removed.

Reply:
This helps to call the respective ams which is passed from mpesa to channel.